clippy_lints/src/transmuting_null.rs

```diff
@@ -0,0 +1,114 @@
+use crate::consts::{constant_context, Constant};
+use crate::utils::{match_qpath, span_lint};
+use if_chain::if_chain;
+use rustc::hir::{Expr, ExprKind};
+use rustc::lint::{in_external_macro, LateContext, LateLintPass, LintArray, LintContext, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use syntax::ast::LitKind;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for transmute calls which would recieve a null pointer.
+    ///
+    /// **Why is this bad?** Transmuting a null pointer is undefined behavior.
+    ///
+    /// **Known problems:** Not all cases can be detected at the moment of this writing.
+    /// For example, variables which hold a null pointer and are then fed to a `transmute`
+    /// call, aren't detectable yet.
+    ///
+    /// **Example:** 
+    /// ```rust
+    /// unsafe {
+    ///     let null_ref: &u64 = std::mem::transmute(0 as *const u64);
+    /// }
```

Comment:
NIT: That's usually how you write `transmute` calls (keep `unsafe` blocks as small as possible):
```rust
let null_ref: &u64 = unsafe { std::mem::transmute(0 as *const u64) };
```

Reply:
I see :D that makes sense

Reply:
Edit added!